File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/model/InstanceGroupType.java
package com.sequenceiq.cloudbreak.api.model;

public enum InstanceGroupType {
    GATEWAY(0), CORE(0);

    private Integer fixedNodeCount;

    InstanceGroupType(Integer fixedNodeCount) {
        this.fixedNodeCount = fixedNodeCount;
    }

    public Integer getFixedNodeCount() {
        return fixedNodeCount;
    }

    public static boolean isGateway(InstanceGroupType type) {
        return GATEWAY.equals(type);
    }

    public static boolean isCoreGroup(InstanceGroupType type) {
        return CORE.equals(type);
    }
}

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/model/InstanceGroupType.kt
package com.sequenceiq.cloudbreak.api.model

enum class InstanceGroupType private constructor(val fixedNodeCount: Int?) {
    GATEWAY(1), CORE(0);


    companion object {

        fun isGateway(type: InstanceGroupType): Boolean {
            return GATEWAY == type
        }

        fun isCoreGroup(type: InstanceGroupType): Boolean {
            return CORE == type
        }
    }
}